examples/benchmark.py

```diff
@@ -81,7 +82,13 @@ def source(t, f0):
 data.set_receiver_pos(receiver_coords)
 data.set_shape(nt, 30)
 # A Forward propagation example
-jit_obj = AcousticWave2D_cg(model1, data, create_dm, nbpml=nbpml, t_order=2, s_order=2)
+
+# path at which numpy memmap file are stored
+cache_disk_path = "/tmp/devito_disk"
+if not os.path.exists(cache_disk_path):
+    os.makedirs(cache_disk_path)
+
+jit_obj = AcousticWave2D_cg(model1, data, create_dm, nbpml=nbpml, t_order=2, s_order=2, disk_path=cache_disk_path)
```

Comment:
@mlange05 I agree with the general principle in your comment above. However, the current reality is that we have an increasingly large number of parameters being passed around in the code. Some being hard-coded, some being picked up from environment variables and some as python parameters. A majority of these are being given "intelligent" default values. However the default values are spread all across the codebase since we have no convention on where the default value needs to be specified. Imagine the devito codebase 10 times as big as it is now and imagine a devito "user" trying to find out where in our codebase the default value for a particular parameter is coming from. Wouldn't it be easier to have it one Singleton config class?